Utilities/rANS/test/test_ransEncodeDecode.cxx

```diff
@@ -0,0 +1,90 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   DecoderSymbol.h
+/// @author Michael Lettrich
+/// @since  2020-04-15
+/// @brief  Test rANS encoder/ decoder
+
+#define BOOST_TEST_MODULE Utility test
+#define BOOST_TEST_MAIN
+#define BOOST_TEST_DYN_LINK
+
+#include <boost/test/unit_test.hpp>
+#include <vector>
+#include <cstring>
+
+#include "../include/librans/SymbolStatistics.h"
+#include "../include/librans/Encoder.h"
+#include "../include/librans/Decoder.h"
+
+BOOST_AUTO_TEST_CASE(test_EncodeDecode32)
+{
+  constexpr char SOURCE[] =
+    R"(Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt 
+	ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo 
+	dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit 
+	amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor 
+	invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et 
+	justo duo dolores et ea rebum. Stet clita kasd gubergren, no 
+	sea takimata sanctus est Lorem ipsum dolor sit amet)";
+
+  constexpr uint PROBABILITY_BITS = 14;
+  constexpr uint32_t PROBABILITY_SCALE = 1 << PROBABILITY_BITS;
+  using coder_t = uint32_t;
+  using stream_t = uint8_t;
+  using source_t = char;
+
+  rans::SymbolStatistics stats{std::begin(SOURCE), std::end(SOURCE)};
+  stats.rescaleFrequencyTable(PROBABILITY_SCALE);
+
+  std::vector<stream_t> encoderBuffer(1 << 20, 0);
+  rans::Encoder<coder_t> encoder{stats, PROBABILITY_BITS};
+  auto ransBegin = encoder.process(encoderBuffer.begin(), encoderBuffer.end(), std::begin(SOURCE), std::end(SOURCE));
+
+  std::vector<source_t> decoderBuffer(stats.getMessageLength(), 0);
+  rans::Decoder<coder_t, source_t> decoder(stats, PROBABILITY_BITS, PROBABILITY_SCALE);
+  decoder.process(decoderBuffer.begin(), ransBegin, stats.getMessageLength());
```

Comment:
perhaps I am missing something, but should not one decode starting from the encoded data, i.e. ``encoderBuffer``.
I think it would be useful to put some comments/guidelines on the steps in this test, e.g. what drives the particular choice of ``PROBABILITY_BITS``, ``stream_t`` etc. 
Is there an example of producing compressed data and storing it with its dictionary (e.g. dumping to a root file, as we discussed)?

Reply:
Thanks for the review. The code is correct but maybe not straight forward to understand. I will improve documentation -  Constants are explained in my CHEP Poster and paper. Plan to improve documentation as soon as the API is more stable.

Reply:
I added some more documentation to the unittest to make sure it is easier to understand and linked my chep poster in the Readme.